rocketpool/node/collectors/node-collector.go

```diff
@@ -255,6 +255,8 @@ func (collector *NodeCollector) Collect(channel chan<- prometheus.Metric) {
 	totalRplSupply := state.NetworkDetails.RPLTotalSupply
 	totalEffectiveStake := collector.stateLocker.GetTotalEffectiveRPLStake()
 	nodeOperatorRewardsPercent := eth.WeiToEth(state.NetworkDetails.NodeOperatorRewardsPercent)
+	nodeWeight := state.GetNodeWeight(state.GetEligibleBorrowedEth(nd), nd.RplStake)
```

Comment:
hmm, before we calculate this, we need to make sure nd.RplStake is above the minimum required (10% of eligible borrowed eth).

If it isn't, weight is 0.

Reply:
See:
```golang
                        // minCollateral := borrowedEth * minCollateralFraction / ratio
                        // NOTE: minCollateralFraction and ratio are both percentages, but multiplying and dividing by them cancels out the need for normalization by eth.EthToWei(1)
                        minCollateral := big.NewInt(0).Mul(eligibleBorrowedEth, s.NetworkDetails.MinCollateralFraction)
                        minCollateral.Div(minCollateral, s.NetworkDetails.RplPrice)

                        // Calculate the weight
                        nodeWeight := big.NewInt(0)
                        if node.RplStake.Cmp(minCollateral) == -1 || eligibleBorrowedEth.Sign() <= 0 {
                                weightSlice[i] = nodeWeight
                                return nil 
                        }

                        nodeWeight.Set(s.getNodeWeight(eligibleBorrowedEth, node.RplStake))
```
in `shared/services/state/network-state.go`

Reply:
Done